lib/qunit-dom.ts

```diff
@@ -12,6 +12,19 @@ QUnit.assert.dom = function(
   target?: string | Element | null,
   rootElement?: Element
```

Comment:
I think you'd want to change the type here to something like

```ts
rootElement?: QuerySelectable
…
interface QuerySelectable {
  querySelector(selector: string): Element,
  querySelectorAll(selectors: string): NodeList
}
```

Reply:
I **do** expect an `Element`, but since the API can be used from JS I can't be sure of it. checking for the two methods seems like a good enough method for us to determine if the passed in thing is an `Element` or not.

Reply:
the reason why I wouldn't want to touch the function signature is that it will cause changes for users that actually do use this library with TypeScript.

Reply:
An interface like the above would make the signature wider actually so should not result in any problems for TS users. I'd argue that if you want to keep the API as it is you should check for an `Element` to be passed – I can't see a reason for deliberately making it so that the signature and code are out of sync.

Reply:
> I'd argue that if you want to keep the API as it is you should check for an Element to be passed

how?

Reply:
something like https://lodash.com/docs#isElement ?

Reply:
IMO the other option would be better though

Reply:
how is https://lodash.com/docs#isElement different/better than the `isValidRootElement()` function in this PR? they're essentially both just guessing, but `isValidRootElement()` at least checks for the things that we actually use.

Reply:
I don't know the details of https://lodash.com/docs#isElement. What I'm basically saying is I think the shape-checking makes sense anyway and I'd reflect that in the TS signature as I cannot see any downsides of doing that and also no upsides of not doing it.

Reply:
> no upsides of not doing it.

I want people to only pass in `Elements` and nothing else. Also, I think `Element` is common terminology and relatively easy to understand, while `QuerySelectable` is something custom that people would have to look up in our type defs.

also the `DomAssertions` class expects `Element | Document`, so if we go with `QuerySelectable` we would either have to cast it to `Element | Document` again, or we'd have to adjust the class to accept `QuerySelectable` instead too. In that case though we'd run into the problem that the signature of the `querySelect()` functions in `QuerySelectable` is not quite the same as in the `ParentNode` interface, which could then cause issues in our TypeScript code.

Reply:
Too much bikeshedding for me - I’m out